### DISCUSSION TITLE
Upgrade to Jackson 2.17

### DISCUSSION THREAD

**Discussion by kozell** [jwtk/jjwt#945] [Q&A] [answered]
Hi,

It seems that the used Jackson version of [2.12.7](https://github.com/jwtk/jjwt/blob/master/pom.xml#L112) in dependency management is quite old (released over 2 years ago), and that version is no longer maintained (as [that was the last](https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.12.7) in that series).

Can we upgrade to the Jackson 2.17.x line?

(If yes, I'll do the PR.)

Thanks,
Joe

**Comment by lhazlewood** [comment]
Hi Joe - thanks for discussing first!

I'd definitely appreciate a PR, but it'll have to pass our build/CI matrix (naturally), and that includes JDK 7, so as long as that all passes, we're fine.

Thanks!

**Comment by kozell** [comment]
JDK 7? 

Yeah, there will be a problem with this upgrade, then...

[https://github.com/FasterXML/jackson-core/blob/jackson-core-2.17.1/pom.xml#L37](https://github.com/FasterXML/jackson-core/blob/jackson-core-2.17.1/pom.xml#L37) says:

```
<!-- 03-May-2022: Change Java compatibility for Jackson-Core 2.14 from Java6 to Java8,
   still use Moditect to get JDK9+ module info support; need newer bundle plugin as well
   (can just defaults from `jackson-parent`)
  -->
```

So this request seems like a big no.

**Reply by lhazlewood** [reply]
We're trying to get off of JDK 7 as soon as feasible, we're just not there yet.

**Reply by lhazlewood** [reply]
P.S. Don't forget you're always able to exclude the JJWT Jackson dependency in maven/gradle and explicitly define the Jackson version you prefer within your project's build configuration.  If you use Jackson in your own application elsewhere (outside of JJWT's needs), this is a best practice anyway since you won't be 'surprised' when you upgrade a library that depends on Jackson.

**Reply by kozell** [reply]
Yes, we already do the exclusion way.

Is there an open issue for the JDK7 removal?

**Reply by lhazlewood** [reply, marked as answer]
Yes, there are a few [issues labeled as jdk8](https://github.com/jwtk/jjwt/issues?q=is%3Aissue+is%3Aopen+label%3Ajdk8).  However, we cannot work on any of them until non-compact JWTs (i.e. full JSON, not comma-delimited) is finished.  That's the last feature required before moving to 1.0, at which time we'll remove JDK 7 compatibility.
